projects/plugins/jetpack/modules/related-posts/jetpack-related-posts.php

```diff
@@ -528,23 +584,23 @@ public function parse_options( $input ) {
 			$input['enabled'] = '1';
 		}
 
-		if ( '1' == $input['enabled'] ) {
-			$current['enabled'] = true;
-			$current['show_headline'] = ( isset( $input['show_headline'] ) && '1' == $input['show_headline'] );
-			$current['show_thumbnails'] = ( isset( $input['show_thumbnails'] ) && '1' == $input['show_thumbnails'] );
-			$current['show_date'] = ( isset( $input['show_date'] ) && '1' == $input['show_date'] );
-			$current['show_context'] = ( isset( $input['show_context'] ) && '1' == $input['show_context'] );
-			$current['layout'] = isset( $input['layout'] ) && in_array( $input['layout'], array( 'grid', 'list' ), true ) ? $input['layout'] : 'grid';
-			$current['headline'] = isset( $input['headline'] ) ? $input['headline'] : esc_html__( 'Related', 'jetpack' );
+		if ( '1' === $input['enabled'] ) {
```

Comment:
In the previous line, we set $input['enabled'] to '1' if any other input options are set, but I'm fine ignoring the rule if you think it's safest.

Reply:
It's all the checks below too.

Reply:
Looks like some of those may return bools. Ignored them for now, unless we want to try and fix them all which might be more work than it's worth.

Reply:
Assuming we're ok with changing the interpretation of '2' (and other non-zero integers and strings) to be true instead of false, we could change them from `isset( $v ) && '1' == $v` to just `! empty( $v )`. But I'm fine with ignoring it for now and considering that later.